dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.ts

```diff
@@ -0,0 +1,110 @@
+   */
+  public static validateAndParseFindResult(results: RawTfmFacility[]): ParsedTfmFacility[] {
+    return results.map((result) => this.validateAndParseFindOneResult(result));
+  }
+
+  /**
+   * Finds the tfm facilities by deal id
+   * @param dealId - The deal id
+   * @returns The found tfm facilities
+   */
+  public static async findByDealId(dealId: string | ObjectId): Promise<RawTfmFacility[]> {
+    const collection = await this.getCollection();
+    return await collection.find({ 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } }).toArray();
+  }
+
+  /**
+   * Finds the tfm facility with the supplied id
+   * @param id - The id of the facility
+   * @returns The found tfm facility
+   */
+  public static async findOneById(id: string | ObjectId): Promise<RawTfmFacility | null> {
+    const collection = await this.getCollection();
+    return await collection.findOne({ _id: { $eq: new ObjectId(id) } });
+  }
+
+  /**
+   * Updates the tfm facility with the supplied id
+   * @param id - The id to search by
+   * @param update - The update to apply
+   * @param [options] - Update options
+   * @returns The document (returned after the update by default)
+   */
+  public static async findOneByIdAndUpdate(
+    id: string | ObjectId,
+    update: UpdateFilter<WithoutId<RawTfmFacility>>,
+    options: FindOneAndUpdateOptions = { returnDocument: 'after' },
+  ) {
+    const collection = await this.getCollection();
+    return await collection.findOneAndUpdate({ _id: { $eq: new ObjectId(id) } }, update, options);
+  }
+
+  /**
+   * Updates the tfm facility with the supplied id
+   * and amendment id
+   * @param id - The facility id
+   * @param amendmentId - The amendment id
+   * @param update - The update to apply
+   * @returns The update result
+   */
+  public static async updateOneByIdAndAmendmentId(id: string | ObjectId, amendmentId: string | ObjectId, update: UpdateFilter<WithoutId<RawTfmFacility>>) {
+    const collection = await this.getCollection();
+    return await collection.updateOne(
+      {
+        _id: { $eq: new ObjectId(id) },
+        'amendments.amendmentId': { $eq: new ObjectId(amendmentId) },
+      },
+      update,
+    );
+  }
+
+  /**
+   * Finds the TFM facilities with the supplied ids
+   * @param ids - The facility ids
+   * @returns The found TFM facilities
+   */
+  public static async findByIds(ids: (string | ObjectId)[]): Promise<RawTfmFacility[]> {
+    const collection = await this.getCollection();
+    return await collection.find({ _id: { $in: ids } }).toArray();
+  }
+
+  /**
+   * Method to perform a custom operation on the TFM
+   * facilities collection
+   * @param operation - The database operation to perform
+   * @returns The result of the supplied operation
+   */
+  public static async custom<TReturn>(operation: (collection: Collection<WithoutId<RawTfmFacility>>) => TReturn | Promise<TReturn>): Promise<TReturn> {
```

Comment:
I think this is better but still feels a bit envious. Happy to go with for now/see what others think

Reply:
Have replaced with an `executeAggregate` method so that the `collection` is no longer exposed